lib/logstash/plugin_mixins/jdbc/scheduler.rb

```diff
@@ -12,6 +12,42 @@ class Scheduler < Rufus::Scheduler
     TimeImpl = defined?(Rufus::Scheduler::EoTime) ? Rufus::Scheduler::EoTime :
                    (defined?(Rufus::Scheduler::ZoTime) ? Rufus::Scheduler::ZoTime : ::Time)
 
+    # @overload
+    def timeout_jobs
+      # Rufus relies on `Thread.list` which is a blocking operation and with many schedulers
+      # (and threads) within LS will have a negative impact on performance as scheduler
+      # threads will end up waiting to obtain the `Thread.list` lock.
+      #
+      # However, this isn't necessary we can easily detect whether there are any jobs
+      # that might need to timeout: only when `@opts[:timeout]` is set causes worker thread(s)
+      # to have a `Thread.current[:rufus_scheduler_timeout]` that is not nil
+      return unless @opts[:timeout]
+      super
+    end
+
+    # @overload
+    def work_threads(query = :all)
+      if query.nil? # our special case from JobDecorator#start_work_thread
```

Comment:
Here we are using the `nil` value as a special value, we are assigning to it a meaning. `work_threads(nil)` is invoked only by the patched methods in `JobDecorator`, could we use something really specific and explicit for this?
Maybe adding a new label `:os_list_threads` or something similar instead of `nil` ?